client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -116,13 +117,57 @@ class CalypsoifyIframe extends Component< Props & ConnectedProps & ProtectedForm
 	componentDidMount() {
 		MediaStore.on( 'change', this.updateImageBlocks );
 		window.addEventListener( 'message', this.onMessage, false );
+
+		this.handleThirdPartyiFrameDomains();
 	}
 
 	componentWillUnmount() {
 		MediaStore.off( 'change', this.updateImageBlocks );
 		window.removeEventListener( 'message', this.onMessage, false );
 	}
 
+	handleThirdPartyiFrameDomains() {
+		// get the domain of the iframed editor so we can check if it is 3rd party domain.
+		// If so we want to initially redirect to that domain to set auth cookie to prevent
+		// 3rd party cookie blocking by browser security
+
+		const { hostname: iFrameDomain, protocol: iFrameProtocol } = url.parse( this.props.iframeUrl );
+		const {
+			hostname: parentDomain,
+			protocol: parentProtocol,
+			port,
+			query: parentQuery,
+		} = url.parse( window.location.href, true );
+		const parentPort = port ? `:${ port }` : '';
+		const firstPartyDomains = [ 'wordpress.com', 'calypso.live', 'calypso.localhost' ];
+		const isThirdPartyDomain = ! some( firstPartyDomains, domain => {
+			return endsWith( iFrameDomain, domain );
+		} );
+
+		if ( isThirdPartyDomain ) {
+			// check query params from parent frame to check if we have already redirected back from 3rd party auth.
+			if ( parentQuery.thirdPartyAuthed ) {
+				// If successfully redirected save to session storage so we don't need to redirect on every editor load
+				sessionStorage.setItem( `calypsoify_${ iFrameDomain }_thirdPartyAuthed`, 'true' );
```

Comment:
The "third party authed" process is officially called "Jetpack SSO". It would be nice to reflect that in the name (both query param and local storage key).

Reply:
Thanks for this clarification - I had '3rd party cookie' stuck in my head while investigating - but narrowing it down to jetpack auth made it easier not just for naming but for the logic to decide whether to run this or not